index.js

```diff
@@ -11,19 +13,44 @@ const levelMapping = ['ansi', 'ansi', 'ansi256', 'ansi16m'];
 // `color-convert` models to exclude from the Chalk API due to conflicts and such
 const skipModels = new Set(['gray']);
 
-function Chalk(options) {
+const styles = Object.create(null);
+
+function applyOptions(obj, options) {
+	options = options || {};
+
 	// Detect level if not set manually
-	this.level = Number(!options || options.level === undefined ? supportsColor.level : options.level);
-	this.enabled = options && 'enabled' in options ? options.enabled : this.level > 0;
+	obj.level = options.level === undefined ? supportsColor.level : options.level;
+	obj.enabled = 'enabled' in options ? options.enabled : obj.level > 0;
+}
+
+function Chalk(options) {
+	// We check for this.template here since calling chalk.constructor()
+	// by itself will have a `this` of a previously constructed chalk object.
+	if (!this || !(this instanceof Chalk) || this.template) {
+		const chalk = {};
+		applyOptions(chalk, options);
+
+		chalk.template = function () {
+			const args = [].slice.call(arguments);
+			return chalkTag.apply(null, [chalk.template].concat(args));
+		};
+
+		Object.setPrototypeOf(chalk, Chalk.prototype);
```

Comment:
Would it be possible to not use `Object.setPrototypeOf` here? Since it's completely unpolyfillable in IE10, it breaks for jest-matchers when trying to use that in the browser.

I know "supporting" older browsers suck, but this is about not failing spectacularily.

Just warpping it in a `if (Object.setPrototypeOf)` might work around the crashing, and then work out all right?

Ref facebook/jest#4074